### PR TITLE
twa(front): Install prettier and add gh action

### PR DESCRIPTION
**Pull request by elenzio9**
In this PR:

- Install Prettier
- Format code
- Add formatting/linting to `twa_frontend_tests` workflow

### CODE REVIEW

File: .github/workflows/twa_frontend_tests.yaml [hunk 1]
```diff
@@ -5,6 +5,31 @@ on:
       - components/crud-web-apps/tensorboards/frontend/**
 
 jobs:
+  test:
```
Comment (kimwnasptd) [inline .github/workflows/twa_frontend_tests.yaml:8]:
```suggestion
  frontend-fromat-linting-check:
```

File: .github/workflows/twa_frontend_tests.yaml [hunk 2]
```diff
@@ -5,6 +5,31 @@ on:
...
+    name: Code format and lint
```
Comment (kimwnasptd) [inline .github/workflows/twa_frontend_tests.yaml:9]:
```suggestion
    name: Check code format and lint```

File: .github/workflows/twa_frontend_tests.yaml [hunk 3]
```diff
@@ -5,6 +5,31 @@ on:
...
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 12
+
+      - name: Format tensorboards web app
```
Comment (kimwnasptd) [inline .github/workflows/twa_frontend_tests.yaml:21]:
```suggestion
      - name: Check frontend code formatting
```

File: .github/workflows/twa_frontend_tests.yaml [hunk 4]
```diff
@@ -5,6 +5,31 @@ on:
...
+        run: |
+          cd components/crud-web-apps/tensorboards/frontend
+          npm i
+          npm run format:check
+
+      - name: Lint tensorboards web app
```
Comment (kimwnasptd) [inline .github/workflows/twa_frontend_tests.yaml:27]:
```suggestion
      - name: Check frontend code linting
```

File: components/crud-web-apps/tensorboards/frontend/package.json [hunk 5]
```diff
@@ -12,7 +12,9 @@
     "test": "ng test",
     "test:prod": "ng test --browsers=ChromeHeadless --watch=false",
     "lint-check": "ng lint",
-    "lint": "ng lint --fix"
+    "lint": "ng lint --fix",
+    "format:check": "prettier --check 'src/**/*.{js,ts,html,scss,css}' || node scripts/check-format-error.js",
```
Comment (kimwnasptd) [inline components/crud-web-apps/tensorboards/frontend/package.json:16]: I see that we don't have this script in TWA, like we do in Katib
https://github.com/kubeflow/katib/tree/master/pkg/new-ui/v1beta1/frontend/scripts

Yet the action didn't fail. Why is that?

Comment (elenzio9) [reply components/crud-web-apps/tensorboards/frontend/package.json:16]: Oh sorry I missed that. Should I add this script in TWA as well?

Comment (kimwnasptd) [reply components/crud-web-apps/tensorboards/frontend/package.json:16]: OK I see what happened. `||` means that the second command (node script in this case) will NOT be run if the first part was successful. But the node script will run if the prettier check fails.

That script is just there to instruct users to run the formatting. Let's include it in this PR

Comment (elenzio9) [reply components/crud-web-apps/tensorboards/frontend/package.json:16]: ACK! I just added it!

File: .github/workflows/twa_frontend_tests.yaml (hunk 1 above)
Comment (kimwnasptd) [inline .github/workflows/twa_frontend_tests.yaml:8]: Could we change this to something like `frontend-format-lint-check`?

**Pull request merged**